src/Domain/Donation.php

```diff
@@ -316,16 +316,24 @@ private function __construct(string $amount, string $currencyCode, PaymentMethod
 
     public static function fromApiModel(DonationCreate $donationData, Campaign $campaign): Donation
     {
-
         $psp = $donationData->psp;
         assert($psp === 'stripe');
 
-        $donation = new self($donationData->donationAmount, $donationData->currencyCode, $donationData->paymentMethodType);
+        $donation = new self(
+            $donationData->donationAmount,
+            $donationData->currencyCode,
+            $donationData->paymentMethodType,
+        );
 
         $donation->setPsp($psp);
         $donation->setUuid(Uuid::uuid4());
         $donation->setCampaign($campaign); // Charity & match expectation determined implicitly from this
+
         $donation->setGiftAid($donationData->giftAid);
+        // `DonationCreate` doesn't support a distinct property yet & we only ask once about GA.
+        $donation->setTipGiftAid($donationData->giftAid);
+        $donation->setTbgShouldProcessGiftAid($donation->getCampaign()->getCharity()->isTbgClaimingGiftAid());
```

Comment:
We already have the campaign, don't need to pull it of the donation:

```suggestion
        $donation->setTbgShouldProcessGiftAid($campaign->getCharity()->isTbgClaimingGiftAid());
```